app/iot/service.py

```diff
@@ -18,32 +18,32 @@ def connect(self) -> None:
     def disconnect(self) -> None:
         ...
 
-    def send_message(self, message_type: MessageType, data: str) -> None:
+    async def send_message(self, message_type: MessageType, data: str) -> None:
         ...
 
 
 class IOTService:
     def __init__(self) -> None:
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
-        device.connect()
+    async def register_device(self, device: Device) -> str:
+        await device.connect()
         device_id = generate_id()
         self.devices[device_id] = device
         return device_id
 
-    def unregister_device(self, device_id: str) -> None:
-        self.devices[device_id].disconnect()
+    async def unregister_device(self, device_id: str) -> None:
+        self.devices.get(device_id).disconnect()
         del self.devices[device_id]
 
-    def get_device(self, device_id: str) -> Device:
-        return self.devices[device_id]
+    async def get_device(self, device_id: str) -> Device:
+        return self.devices.get(device_id)
```

Comment:
The `get_device` method could potentially return `None` if the device_id is not found in the devices dictionary. You should handle this case appropriately to prevent any potential issues downstream.